pandas/core/indexes/base.py

```diff
@@ -3244,9 +3244,9 @@ def join(self, other, how='left', level=None, return_indexers=False,
         elif how == 'right':
             join_index = other
         elif how == 'inner':
-            join_index = self.intersection(other, sort=sort)
+            join_index = self.intersection(other, sort=False)
```

Comment:
On Master ``intersection`` does not generally sort by default so I have put ``sort=False`` here for now for compatibility reasons so that the current behaviour of ``join(..., how='inner')`` does not change for now. We could change this in future so that the ``intersection`` function here will simply take the ``sort`` parameter passed to the ``join`` function.

Reply:
can you add a TODO about this; this might be #24730